test/types/fastify.test-d.ts

```diff
@@ -12,9 +13,71 @@ expectType<FastifyInstance<http.Server, http.IncomingMessage, http.ServerRespons
 // https server
 expectType<FastifyInstance<https.Server, http.IncomingMessage, http.ServerResponse>>(fastify({ https: {} }))
 // http2 server
-expectType<FastifyInstance<http2.Http2Server, http2.Http2ServerRequest, http2.Http2ServerResponse>>(fastify({ http2: true }))
+expectType<FastifyInstance<http2.Http2Server, http2.Http2ServerRequest, http2.Http2ServerResponse>>(fastify({ http2: true, http2SessionTimeout: 1000 }))
 expectType<FastifyInstance<http2.Http2SecureServer, http2.Http2ServerRequest, http2.Http2ServerResponse>>(fastify({ http2: true, https: {}}))
 expectType<LightMyRequestChain>(fastify({ http2: true, https: {}}).inject())
 
 expectError(fastify<http2.Http2Server>({ http2: false })) // http2 option must be true
 expectError(fastify<http2.Http2SecureServer>({ http2: false })) // http2 option must be true
+
+// server options
+expectAssignable<FastifyInstance>(fastify({ http2: true }))
+expectAssignable<FastifyInstance>(fastify({ ignoreTrailingSlash: true }))
+expectAssignable<FastifyInstance>(fastify({ connectionTimeout: 1000 }))
+expectAssignable<FastifyInstance>(fastify({ keepAliveTimeout: 1000 }))
+expectAssignable<FastifyInstance>(fastify({ pluginTimeout: 1000 }))
+expectAssignable<FastifyInstance>(fastify({ bodyLimit: 100 }))
+expectAssignable<FastifyInstance>(fastify({ maxParamLength: 100 }))
+expectAssignable<FastifyInstance>(fastify({ disableRequestLogging: true }))
+expectAssignable<FastifyInstance>(fastify({ requestIdLogLabel: 'request-id' }))
+expectAssignable<FastifyInstance>(fastify({ onProtoPoisoing: 'error' }))
+expectAssignable<FastifyInstance>(fastify({ onConstructorPoisoning: 'error' }))
+expectAssignable<FastifyInstance>(fastify({ logger: true }))
+expectAssignable<FastifyInstance>(fastify({
```

Comment:
for what its worth, I believe we are already testing the logger option in the logger type def test file. these are redundant here. Nice to have for sure but not necessary.

In order to get them to pass you need to pass the same logger type to the fastify instance generic paramter (thus also passing the http generics). Ik this is not ideal but its whats necessary

Reply:
Thanks for the help. I decided to keep the tests as they provide examples of how to overload logger options with various types and they also helped fix some bugs in the definitions for `FastisfyLoggerInstance['child']` and `FastifyLoggerOptions`.